instrumentation/tomcat/tomcat-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/tomcat/common/TomcatHelper.java

```diff
@@ -66,4 +67,15 @@ public void attachResponseToRequest(Request request, Response response) {
       servletHelper.setAsyncListenerResponse(servletRequest, servletResponse);
     }
   }
+
+  static String messageBytesToString(MessageBytes messageBytes) {
+    // on tomcat 10.1.0 MessageBytes.toString() has a side effect. Calling it caches the string
```

Comment:
Is this issue only happening on Tomcat 10? I am recently experiencing an issue with Java agent whose version is lower than 1.9.0. And the only change we did recently is bumping up the tomcat version from 8.5.84 to 8.5.87, not sure if it is possible for you to share when that MessageBytes.toString() issue was introduced to tomcat?

Reply:
Tomcat authors sometimes backport changes from newer release branches to older ones. I believe that this https://github.com/apache/tomcat/commit/8e40ef3ae8d6f27324cc6370f63809a3d1870f8f is the commit that introduces the change in 8.x branch.